lib/hanami/model/association.rb

```diff
@@ -0,0 +1,23 @@
+require 'hanami/model/associations/has_many'
+
+module Hanami
+  module Model
+    # Association factory
+    #
+    # @since x.x.x
+    # @api private
+    class Association
+      # Instantiate an association
+      #
+      # @since x.x.x
+      # @api private
+      def self.new(repository, target, subject)
+        case repository.root.associations[target]
+        when ROM::SQL::Association::OneToMany then Associations::HasMany
+        else
+          raise 'unsupported association'
+        end.new(repository, repository.root.name.to_sym, target, subject)
```

Comment:
For now I just changed it to `Association.build` (see https://github.com/hanami/model/pull/334/commits/480d1f48a2e1d9297ae5cc514c6cca3b93eb263b). This is still experimental code that needs to be revised later. It will get a better shape later 😉